minetest/minetest
# Allow overriding fonts via media files

Implements #10416 in its most basic form, allowing you to override the fonts that are currently used via media files. Does not include font escape sequences.

## To do

This PR is ready for review.

- Decide on the font file format(s) we want to support:
  - Opted for TTF (for convenience) and WOFF2 (compression, tends to be 2x-3x smaller) for now.
  - Feel free to propose more (or less).
- Basic security considerations:
  - Probably not terrible since that's a FreeType problem and big browsers use FreeType.
  - Looking at the CVEs for FreeType, not much (critical) vulnerabilities seem to have been found since 2015: https://www.cvedetails.com/vendor/4535/Freetype.html
  - Generally I think this is not very different in principle to relying on the safety of other dependencies, e.g. libpng, libjpeg etc. which are used to load images, though indeed font file formats may tend to be a bit more problematic.
  - If someone has a good suggestion for a font format that is demonstrably less problematic yet not comparatively obscure or inefficient, please share it.

## How to test

* Enable this mod: 
[fonts.zip](https://github.com/user-attachments/files/18387177/fonts.zip)

* Join game.
* Observe different fonts. Use `/show_fonts` if you want to see all 8 different fonts.
* Leave. Observe normal fonts.

![Screenshot from 2025-01-10 02-27-17](https://github.com/user-attachments/assets/2c348865-0288-476e-a1c4-fcc65c3eb18d)

If you're feeling funny you can also test with dynamic media.
